pkg/endpoint/policy_test.go

```diff
@@ -38,3 +53,163 @@ func (s *EndpointSuite) TestUpdateVisibilityPolicy(c *check.C) {
 	})
 	c.Assert(ep.visibilityPolicy, check.IsNil)
 }
+
+// This test fuzzes the incremental update engine from an end-to-end perspective
+// to ensure we don't ever miss an incremental update.
+//
+// It works by simulating a "churning" IPcache that is constantly allocating new identities.
+// There is a single policy that -- funnily enough -- selects all of the new identities.
+// We then continuously simulate endpoint regeneration and ensure the computed policy contains
+// all the generated identities.
+//
+// By default, we test 1000 identities, which should take less than 10 seconds. If this test fails,
+// please bump the factor to something massive and start debugging :-).
+func TestIncrementalUpdatesDuringPolicyGeneration(t *testing.T) {
+	const testfactor = 1000 // bump this to stress-test
+
+	policy.SetPolicyEnabled("always")
```

Comment:
Incidentally, is this statically defined and could it impact later test runs? :thinking: 

I know in the past we had some weird defer logic for global settings like this to revert the setting back to avoid affecting other tests. I guess if the tests are passing it doesn't directly apply, but it could plausibly trip someone up in future.

Reply:
you're right, this should be restored.